crates/dies-world/src/filter/matrix_gen.rs

```diff
@@ -0,0 +1,129 @@
+use nalgebra as na;
```

Comment:
I wonder, could these types be implemented as utility functions? Generally, I try to avoid introducing traits unless they make the code cleaner. Here I'm not sure it is necessary. What do you think?

Reply:
I think the main problem is [here](https://github.com/DelftMercurians/Dies/blob/99e78b88d7ccdbd8aa0e20fcbfdc118e644b103b/crates/dies-world/src/filter/kalman.rs#L109-L111). **The goal is to have a transformation matrix A and noise matrix Q whose content can be changed based on `t`.** Different modeling requires different shapes of A and Q. But if we made them as functions I don't know how to have a single variable `A`(Q) that can fit all kinds of needs.
One way I can think of is to 
1. for Q we get rid of those fancy(?) noise matrices, just use a diagonal matrix with some std. The structure of Q doesn't affect that much anyway
2. we store A in a higher level(Players/Ball), and feed it everytime as a parameter of `update`'


But 2 just gives me a feeling that the filter relies on smth it doesn't have to and the tracker maintenance smth it doesn't have to. What do you think